include/mbedtls/x509_crt.h

```diff
@@ -47,14 +48,85 @@ extern "C" {
  * \{
  */
 
+typedef struct mbedtls_x509_crt_frame
+{
+    /* Keep these 8-bit fields at the front of the structure to allow them to
+     * be fetched in a single instruction on Thumb2; putting them at the back
+     * requires an intermediate address calculation. */
+
+    uint8_t version;                        /**< The X.509 version. (1=v1, 2=v2, 3=v3)                          */
+    unsigned char ca_istrue;                /**< Optional Basic Constraint extension value:
+                                             *   1 if this certificate belongs to a CA, 0 otherwise.            */
+    unsigned char max_pathlen;              /**< Optional Basic Constraint extension value:
+                                             *   The maximum path length to the root certificate.
+                                             *   Path length is 1 higher than RFC 5280 'meaning', so 1+         */
+    unsigned char ns_cert_type;             /**< Optional Netscape certificate type extension value:
+                                             *   See the values in x509.h                                       */
+
+    unsigned int key_usage;                 /**< Optional key usage extension value: See the values in x509.h   */
```

Comment:
@andresag01 Yes, definitely, I tried to make `mbedtls_x509_crt_frame` as compact as I can, and e.g. `ext_types` must be `uint32_t` because `NS Cert Type` has ID `1 << 16`. However, `key_usage` can indeed be made smaller, I think, because its largest value is `0x8000` - I must have mistakenly thought at some point that this would need 32 bits.

I resized `key_usage` to `uint16_t` and reordered it with `ext_types` to avoid padding and have it pack nicely with `sig_md` and `sig_pk`. Thanks!

Reply:
Fixed in 4df4e5f.